microsoft/testsuites/performance/common.py

```diff
@@ -160,9 +160,20 @@ def perf_tcp_latency(
     return latency_perf_messages
 
 
-def perf_tcp_pps(environment: Environment, test_type: str) -> None:
-    client = cast(RemoteNode, environment.nodes[0])
-    server = cast(RemoteNode, environment.nodes[1])
+def perf_tcp_pps(
+    environment: Environment,
+    test_type: str,
+    server: Optional[RemoteNode] = None,
+    client: Optional[RemoteNode] = None,
+) -> None:
+    if server is None:
```

Comment:
Explain more in which scenario it's set outside, and when one of them are set. If it's not right to set one of them outside, it needs to check either both of them None or both of them are set.